src/openms/source/CHEMISTRY/EmpiricalFormula.cpp

```diff
@@ -643,16 +643,9 @@ namespace OpenMS
       return formula_.size() < rhs.formula_.size();
     }
 
-    // both maps have same size
-    auto it = formula_.begin();
-    auto rhs_it = rhs.formula_.begin();
-    for (; it != formula_.end(); ++it, ++rhs_it)
-    {
-      if (*(it->first) != *(rhs_it->first)) return *(it->first) < *(rhs_it->first); // element
-      if (it->second != rhs_it->second) return it->second < rhs_it->second; // count
-    }
+    if (charge_ != rhs.charge_) return charge_ < rhs.charge_;
```

Comment:
changes semantics / ordering
did you try the simpler:
return std::tie(formula_size(), charge_, formula_) < std::tie(rhs.formula_size(), rhs.charge_, rhs.formula_)
should be optimized to the same code

Reply:
std::tie() takes only lvalue references as far as I understand. Your approach works, when leaving the formula.size() out of the tie. Is 
`if (formula_.size() != rhs.formula_size()) return formula_.size() < rhs.formula_.size();
return std::tie(charge_, formula_) < std::tie(rhs.charge_, rhs.formula_);`
more favourable than the current solution?

Reply:
yes I was thinking if it makes sense to have formula_.size() begin tested in formula_ < rhs.formula_
but I guess that is out of scope here. so feel free to leave it as is